hal/README.md

```diff
@@ -134,11 +134,41 @@ The object provided to the Mustache template for automatically-generated documen
 }
 ```
 
+## Utilities `@azure-iot/hal/util`
+
+### Server
+Server-specific utility methods can be accessed by calling the utility import as a function on the server object, eg. `util(server).method(...)`.
+
+#### `links(rel, [params])`
+Returns all of the links for the provided rel as an array of `Hal.Link` values.
+* `rel` [string | LinkRelation]: The rel for which to retrieve the links.
+* `overrides` [hal.Overrides] *(optional)*: Allows you to override the default behavior of the rel, as described above.
+
+#### `href(rel, [params])`
+A shorthand to return a single `string` href for the rel; if there are multiple routes present for the rel, it will simply select the first one, preferring discoverable routes.
+* `rel` [string | LinkRelation]: The rel for which to retrieve the href.
+* `params` [any] *(optional)*: A parameter object used to populate the href, if it is a templated route.
+
+### Generic
+Generic utility methods can be accessed directly from the utility import, eg. `util.method()`.
+
+#### `template(href, params)`
```

Comment:
why is this template, when what is essentially equivalent functionality is called href above?